vtex-apps/product-list-context
# useProductImpression

#### What does this PR do? \*

Implement the `useProductImpression` hook which is responsible for checking the `toBeImpressed` prop of the `ProductListContext` and sending the `productImpression` event when the time is right. To better reflect this new feature, the actions and state props were renamed to better represent their roles.

**IS THIS A BREAKING CHANGE!?** 
Yes, but this app is new and I believe nobody is using it yet >:D

#### How to test it? \*

[Workspace](https://iaronaraujo--storecomponents.myvtex.com/)
then on console type `pixelManagerEvents` to check the events that are being sent when you see the shelf.

#### To be done before this PR is ready for merge:
- type things
- maybe rename more stuff
- changelog
- readme
